app/population/index.js

```diff
@@ -9,23 +9,32 @@ const {
 } = require('../../common/middleware/collection')
 const wizard = require('../../common/middleware/unique-form-wizard')
 
-const { BASE_PATH, MOUNTPATH, DAILY_PATH } = require('./constants')
-const { dashboard, daily } = require('./controllers')
+const { BASE_PATH, MOUNTPATH, DAILY_PATH, WEEKLY_PATH } = require('./constants')
+const { dashboard, daily, weekly } = require('./controllers')
 const { editFields } = require('./fields')
 const {
   redirectBaseUrl,
-  setBreadcrumb,
+  setLocationFreeSpaces,
   setPopulation,
-  setResultsAsPopulationTables,
+  setBreadcrumb,
+  setRegionalOrganisationSwitcher,
+  setResultsAsFreeSpacesTables,
+  setResultsAsFreeSpacesAndTransfersTables,
 } = require('./middleware')
 const { editSteps } = require('./steps')
 
 router.param('date', setDateRange)
-router.param('locationId', setPopulation)
```

Comment:
Are these two pieces of middleware required for every route?

Might be cleaner to use `router.use()` if so:

```javascript
...

router.use(setLocationFreeSpaces , setPopulation)

dailyRouter.get('/', setBreadcrumb, daily)

...
```

And I was going to ask if `setBreadcrumb` could also be used for all routes? So could be:

```javascript
...

router.use(setLocationFreeSpaces , setPopulation, setBreadcrumb)

dailyRouter.get('/', daily)

...
```

Reply:
There might be an ordering issue - but I'll give it a try, as that'd be less repeated code everywhere

Reply:
This was a good suggestion, but I wasn't able to get this to work successfully - there must be some instantiation order between the params and the various middleware.

This will have to be a future refactoring.